subprojects/gradle/signer/src/main/kotlin/com/avito/plugin/SignArtifactTask.kt

```diff
@@ -51,6 +52,13 @@ abstract class SignArtifactTask @Inject constructor(objects: ObjectFactory) : De
             .connectTimeout(10L, TimeUnit.SECONDS)
             .writeTimeout(40L, TimeUnit.SECONDS)
             .readTimeout(40L, TimeUnit.SECONDS)
+            .addInterceptor(
+                RetryInterceptor(
+                    retries = 3,
+                    allowedMethods = listOf("GET", "POST"),
```

Comment:
should we simplify API in scenarious when we want to retry all requests?

Reply:
I found only 4 usages and in all cases add POST because it's only GET by default. 
It seems we care only about response codes (5xx) but not about methods.

I propose to retry all methods by default.
@dsvoronin , @sboishtyan , what do you think?

Reply:
I agree

Reply:
RetryInterceptor is general purpose interceptor, anyone can use it now
in REST only GET methods can be cached (or be idempotent),
and retry POST methods is generally a mistake

that's the reason behind defaults

Why remove this configurability? For what reason? What does we gain in exchange?

Reply:
> retry POST methods is generally a mistake

Current API doesn't give an ability to have such configurations:

- GET <---> { codes }
- POST <---> { codes }

See how clients use it. 
Clients want to retry 5xx errors.  They configure POST because they use POST methods. 

> Why remove this configurability?

No one is talking about removing configurations. 
The proposal is only to change defaults for `allowedMethods` - include all methods by default.
Because this is how it's used now.